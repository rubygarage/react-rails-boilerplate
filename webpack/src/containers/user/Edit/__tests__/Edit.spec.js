import React from 'react';
import UserEditContainer from 'containers/user/Edit';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

describe('UserEdit container', () => {
  const store = configureStore()({
    entities: { users: { 1: { id: '1', username: 'name' } } },
    form: { userEdit: { values: { avatar: [] } } },
    signin: { currentUser: { users: ['1'] } },
  });
  store.dispatch = jest.fn();

  const defaultProps = {
    match: {
      params: {
        id: '1',
      },
    },
    store,
  };

  const container = shallow(<UserEditContainer {...defaultProps} />);

  it('renders UserEdit component', () => {
    expect(container).toMatchSnapshot();
  });

  it('map state and dispatch to props', () => {
    expect(container.props()).toEqual(expect.objectContaining({
      avatarRemovalHandler: expect.any(Function),
      getUserAction: expect.any(Function),
      id: '1',
      user: { id: '1', username: 'name' },
      avatarFieldValue: [],
      match: { params: { id: '1' } },
    }));
  });
});
