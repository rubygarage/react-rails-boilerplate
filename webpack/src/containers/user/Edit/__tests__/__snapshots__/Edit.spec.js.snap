// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEdit container renders UserEdit component 1`] = `
<InjectIntl(ReduxForm)
  avatarFieldValue={Array []}
  avatarRemovalHandler={[Function]}
  getUserAction={[Function]}
  id="1"
  match={
    Object {
      "params": Object {
        "id": "1",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "id": "1",
      "username": "name",
    }
  }
/>
`;
